dlopen-notes.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: CC0-1.0
+
+# Parse .note.dlopen notes from ELF files and prints 'soname priority' pairs on separate lines.
+
+import json
+import sys
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import NoteSection
+
+def dlopen_notes_parse(elf):
+    sonames = {}
+
+    with open(elf, 'rb') as f:
```

Comment:
`with` is actually not useful here, since (read-only) open files are just another resource managed automatically by python and the system.